### ISSUE TITLE
Backward compatibility is broken when replaying 32bit trace files with portability table

### ISSUE THREAD

**Issue by justuswang** [LunarG/VulkanTools#421]
@davidlunarg, looks like #379 breaks the backward compatibility of portability table in replaying 32bit trace file because the portability table was using "size_t" but now it is using "uint64_t".

Is it intended or a bug?

**Comment by davidlunarg** [comment]
Yes, it does break compatibility, and I was aware of it when I made the change. I have not heard of anyone that has a collection of 32 bit Vulkan traces, only of folks that have 64 bit Vulkan traces. Given there are so few 32 bit Vulkan traces, I didn't think breaking 32 bit compatibility would be an issue. Maybe I just found someone that does have a collection of 32 bit trace files. :-)

A trace file can be converted from the old format to the new format by replaying it with an older version of vkreplay and recording the playback with a new version of vktrace.

Note that the change from size_t to uint64_t was needed in order to support 32 bit trace files larger than ~4G.
